File: src/main/java/br/com/thiagoodev/portfolio/infrastructure/persistence/repositories/ProjectsRepository.java
package br.com.thiagoodev.portfolio.infrastructure.persistence.repositories;

import br.com.thiagoodev.portfolio.domain.entities.Project;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface ProjectsRepository extends JpaRepository<Project, Long> {
    public List<Project> findAllByOrderByUpdatedAtDesc();
}

File: services/projects_sync/internal/domain/repositories/projects_repository.go
package repositories

import (
	"context"

	"github.com/cthiagoodev/thiagoodev-portfolio/services/projects_sync/internal/domain/entities"
)

type ProjectsRepository interface {
	GetAll(ctx context.Context) ([]entities.Project, error)
	ResetAndCreateAll(ctx context.Context, projects []entities.Project) error
}
